js/packages/teams-ai/src/planners/AssistantsPlanner.ts

```diff
@@ -0,0 +1,309 @@
+        }
+    }
+
+    private generatePlanFromTools(state: TState, required_action: OpenAI.Beta.Threads.Runs.Run.RequiredAction): Plan {
+        const plan: Plan = { type: 'plan', commands: [] };
+        const tool_map: {[key: string]: string} = {};
+        required_action.submit_tool_outputs.tool_calls.forEach(tool_call => {
+            tool_map[tool_call.function.name] = tool_call.id;
+            plan.commands.push({
+                type: 'DO',
+                action: tool_call.function.name,
+                parameters: JSON.parse(tool_call.function.arguments)
+            } as PredictedDoCommand);
+        });
+        state.setValue(SUBMIT_TOOL_OUTPUTS_MAP, tool_map);
+        return plan;
+    }
+
+    private async generatePlanFromMessages(thread_id: string, last_message_id: string): Promise<Plan> {
+        // Find the new messages
+        const messages = await this.listMessages(thread_id);
+        const newMessages: OpenAI.Beta.Threads.Messages.ThreadMessage[] = [];
+        for (let i = 0; i < messages.data.length; i++) {
+            const message = messages.data[i];
+            if (message.id == last_message_id) {
+                break;
+            } else {
+                newMessages.push(message);
+            }
+        }
+
+        // Convert the messages to SAY commands
+        const plan: Plan = { type: 'plan', commands: [] };
+        newMessages.forEach(message => {
+            message.content.forEach(content => {
+                if (content.type == 'text') {
+                    plan.commands.push({
+                        type: 'SAY',
+                        response: (content as any).text.value
+                    } as PredictedSayCommand);
+                }
+            });
+        });
+        return plan;
+    }
+
+    private ensureAssistantsState(state: TState): AssistantsState {
+        if (!state.hasValue(this._options.assistants_state_variable!)) {
+            state.setValue(this._options.assistants_state_variable!, {
+                thread_id: null,
+                run_id: null,
+                last_message_id: null
+            } as AssistantsState);
+        }
+
+        return state.getValue(this._options.assistants_state_variable!);
+    }
+
+    private updateAssistantsState(state: TState, assistants_state: AssistantsState): void {
+        state.setValue(this._options.assistants_state_variable!, assistants_state);
+    }
+
+    private async ensureThreadCreated(state: TState, input: string): Promise<string> {
+        const assistants_state = this.ensureAssistantsState(state);
+        if (assistants_state.thread_id == null) {
+            const thread = await this.createThread({});
+            assistants_state.thread_id = thread.id;
+            this.updateAssistantsState(state, assistants_state);
+        }
+
+        return assistants_state.thread_id;
+    }
+
+    private async blockOnInprogressRuns(thread_id: string): Promise<void> {
```

Comment:
I am undecided between `blockOnInprogressRuns` and `blockOnInProgressRuns`

Reply:
I renamed it